build-tools/packages/build-cli/src/test/commands/generate/assertTags.test.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync } from "node:fs";
+import path from "node:path";
+import { cosmiconfig } from "cosmiconfig";
+import { describe, it } from "mocha";
+
+/**
+ * Loader for .mjs (ESM) config files.
+ * Required for cosmiconfig v9+ which removed default .mjs support.
+ * This duplicates the loader from assertTags.ts to test it works correctly.
+ */
+async function mjsLoader(filepath: string): Promise<unknown> {
+	const module = await import(filepath);
+	return module.default;
+}
+
+describe("generate:assertTags", () => {
+	describe("cosmiconfig .mjs loader", () => {
+		const configName = "assertTagging";
+		const searchPlaces = [`${configName}.config.mjs`];
+
+		it("loads .mjs config files with custom loader", async () => {
+			const config = cosmiconfig(configName, {
+				searchPlaces,
+				loaders: {
+					".mjs": mjsLoader,
+				},
+			});
+
+			// Test with packages/test directory which has an assertTagging.config.mjs
+			// Navigate up from build-tools/packages/build-cli to root
+			const repoRoot = path.resolve(process.cwd(), "../../..");
+			const testDir = path.join(repoRoot, "packages/test");
+
+			// Skip test if the directory doesn't exist (e.g., in isolated test environment)
+			if (!existsSync(testDir)) {
+				console.log(`Skipping test: ${testDir} does not exist`);
+				return;
+			}
```

Comment:
Can we create a mock env or fixtures instead of relying on the layout of the repo?